### DISCUSSION TITLE
0.74: Int32 | null generates `Nullable Integer` on java side, but `Ljava/lang/Double` on c++ side.

### DISCUSSION THREAD

**Discussion by mfazekas** [reactwg/react-native-new-architecture#198] [Q&A] [answered]
In RN 0.74 we started to see `Error: Exception in HostFunction: no non-static method "Lcom/rnmapbox/rnmbx/components/mapview/NativeMapViewModule;.setHandledMapChangedEvents(Ljava/lang/Double;Lcom/facebook/react/bridge/ReadableArray;Lcom/facebook/react/bridge/Promise;)V`

We implement the function as 
```kotlin
    typealias ViewRefTag = Int
    ...
   override fun setHandledMapChangedEvents(
        viewRef: ViewRefTag?,
        events: ReadableArray,
        promise: Promise
    ) {
```

### Codespec: Int32
Our [maps/src/specs/NativeMapViewModule.ts](https://github.com/rnmapbox/maps/blob/34db17e58f5b8c23ed55d0820b423f5a2cd30850/src/specs/NativeMapViewModule.ts#L41-L44) has:

```ts
setHandledMapChangedEvents: ( viewRef: Int32 | null,  events: ReadonlyArray<string>) => Promise<Object>; 
```

### specs-generated.cpp: java.lang.Double

Codegen generated cpp: `android/build/generated/source/codegen/jni/rnmapbox_maps_specs-generated.cpp` has `Ljava/lang/Double`
```cpp
static facebook::jsi::Value __hostFunction_NativeMapViewModuleSpecJSI_setHandledMapChangedEvents(facebook::jsi::Runtime& rt, TurboModule &turboModule, const facebook::jsi::Value* args, size_t count) {
  static jmethodID cachedMethodId = nullptr;
  return static_cast<JavaTurboModule &>(turboModule).invokeJavaMethod(rt, PromiseKind, "setHandledMapChangedEvents", "(Ljava/lang/Double;Lcom/facebook/react/bridge/ReadableArray;Lcom/facebook/react/bridge/Promise;)V", args, count, cachedMethodId);
}
```

### ModuleSpec.java: java.lang.Integer

Codegen generate-specs-cli.js generated [maps/android/src/main/old-arch/com/rnmapbox/rnmbx/NativeMapViewModuleSpec.java](https://github.com/rnmapbox/maps/blob/34db17e58f5b8c23ed55d0820b423f5a2cd30850/android/src/main/old-arch/com/rnmapbox/rnmbx/NativeMapViewModuleSpec.java#L78-L80) has as:

```java
public abstract void setHandledMapChangedEvents(@Nullable Integer viewRef, ReadableArray events, Promise promise); 
```

See https://github.com/rnmapbox/maps/pull/3527

**Comment by cortinico** [comment, marked as answer]
This looks like a bug. Can you open an issue on facebook/react-native with "Type: New Architecture".

Also is this a regression from 0.73 to 0.74? @mfazekas
